tests/context.py

```diff
@@ -46,7 +46,12 @@ def get_elastic_ip(self, retry=3, interval=1):
             retry -= 1
 
         for serv in services.items:
-            if serv.metadata.name == 'elasticsearch-master':
+            # Amit: following a bug - sometimes elasticsearch-master service will be returned before being assigned with
+            # an IP address, in that case run this function again recursively.
+            if serv.metadata.name == 'elasticsearch-master' and not serv.status.load_balancer.ingress:
+                time.sleep(interval)
+                return self.get_elastic_ip()
```

Comment:
Can we prevent an infinite loop here? E.g., decrement the `retry` counter each time you make a recursive call, something like that?